src/bluetooth_numbers/reverse_lookup.py

```diff
@@ -0,0 +1,127 @@
+"""Reverse lookup class to find UUIDs by their description."""
+from __future__ import annotations
+
+from typing import Literal, NamedTuple, Sequence, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from uuid import UUID
+
+from bluetooth_numbers import characteristic, company, descriptor, oui, service
+
+LOGIC = Literal["OR", "AND", "SUBSTR"]
+UUID_TYPE_DEFAULT: Sequence[str] = (
+    "characteristic",
+    "company",
+    "descriptor",
+    "oui",
+    "service",
+)
+
+
+class Match(NamedTuple):
+    """Named tuple to hold a UUID and its description."""
+
+    uuid: str | UUID | int
+    description: str
+    uuid_type: str
+
+
+class ReverseLookup:
+    """Reverse lookup class to find UUIDs by their description.
+
+    Examples:
+            >>> from bluetooth_numbers.reverse_lookup import ReverseLookup, Match
+            >>> rl = ReverseLookup()
+            >>> matches = rl.lookup("Cycling Power")
+            >>> Match('00:05:5A', 'Power Dsine Ltd.', 'oui') in matches
+            True
+            >>> rl.lookup("Cycling Power", logic="AND")
+            {Match(uuid=6168, description='Cycling Power', uuid_type='service'),
+             Match(uuid=10851, description='Cycling Power Measurement',
+             uuid_type='characteristic'),
+             Match(uuid=10852, description='Cycling Power Vector',
+             uuid_type='characteristic'),
+             Match(uuid=10853, description='Cycling Power Feature',
+             uuid_type='characteristic'),
+             Match(uuid=10854, description='Cycling Power Control Point',
+             uuid_type='characteristic')}
```

Comment:
Sets don't have an ordering, so this output isn't guaranteed to be the same. You can see it by running `tox -e doctests`. Maybe better to illustrate this feature by doing a `Match(...) not in matches`?